prov/efa/src/rdm/efa_rdm_pke.c

```diff
@@ -614,43 +614,51 @@ int efa_rdm_pke_write(struct efa_rdm_pke *pkt_entry)
  *
  * @param[in] pke_vec	packet entries that contains information of receive buffer
  * @param[in] pke_cnt	Number of packet entries to post receive requests for
+ * @param[in] flags  	user supplied flags passed to fi_recv
  * @return		0 on success
  * 			On error, a negative value corresponding to fabric errno
  */
 ssize_t efa_rdm_pke_recvv(struct efa_rdm_pke **pke_vec,
-			  int pke_cnt)
+			  int pke_cnt, uint64_t flags)
 {
 	struct efa_rdm_ep *ep;
 	struct ibv_recv_wr *bad_wr;
 	struct ibv_qp *qp;
 	int i, err;
+	size_t wr_index;
 
 	assert(pke_cnt);
 
 	ep = pke_vec[0]->ep;
 	assert(ep);
 
 	for (i = 0; i < pke_cnt; ++i) {
-		ep->base_ep.efa_recv_wr_vec[i].wr.wr_id = (uintptr_t)pke_vec[i];
-		ep->base_ep.efa_recv_wr_vec[i].wr.num_sge = 1;
-		ep->base_ep.efa_recv_wr_vec[i].wr.sg_list = ep->base_ep.efa_recv_wr_vec[i].sge;
+		wr_index = ep->base_ep.efa_recv_wr_index + i;
```

Comment:
It may be better to have `wr_index = ep->base_ep.efa_recv_wr_index` before this loop, and make `wr_index++` in the loop. WDYT?

Reply:
I think it is better to increment efa_recv_wr_index only once outside the loop. It is less error prone.

Reply:
You can still update `efa_recv_wr_index` once outside the loop. My thoughts are 

```
wr_index = ep->base_ep.efa_recv_wr_index

for (i = 0 ; i< pkt_cnt; i++) {
...
recv_wr = &ep->base_ep.efa_recv_wr_vec[wr_index];
... // do the work
wr_index++;
}
ep->base_ep.efa_recv_wr_index = wr_index;
```

which looks cleaner.

Reply:
Updated